source/docs/software/getting-started/wpilib-setup.rst

```diff
@@ -0,0 +1,208 @@
+      2. Drag the contents of WPILIB\_Linux-2019.1.1 directory to ~/frc2019 or run ``$ mv -v WPILib_Linux-2019.1.1/* ~/frc2019``
+
+      .. figure:: images/linux/toolsupdater.png
+         :alt: Tools Updater
+
+    **Running Tools Updater**
+
+      To extract the WPILib tools (Dashboards, Robot Builder, etc.), run: \* ``$ cd ~/frc2019/tools`` \* ``$ python3 ToolsUpdater.py``
+
+    **Installing the extensions for WPILib VS Code**
+
+      1. Start VSCODE ($ code or search "Visual Studio Code" in your application launcher)
+      2. *Control-Shift-P* to bring up the command palette, type "Install from VSIX"
+      3. Select "Extensions: Install from VSIX"
+      4. Navigate to ``~/frc2019/vsCodeExtensions`` and select Cpp.vsix
+      5. Repeat for JavaLang.vsix, JavaDeps.vsix, JavaDebug.vsix, and WPILib.vsix in that order
+
+      .. figure:: images/linux/install-vsix.png
+          :alt: Install VSIX
+
+      .. figure:: images/linux/vsix-files.png
+          :alt: VSIX Files
+
+    **Setting up VSCode to use Java 11**
+    
+      The WPILib installation includes a JDK, however you need to point VS Code at where it is. To do this:
+
+       1. Open VS Code
+       2. Press *Ctrl-Shift-P* and type *WPILib* or click on the WPILib icon in the top right to open the WPILib Command Palette
+       3. Begin typing *Set VS Code Java Home to FRC Home* and select that item from the dropdown.
+
+      .. figure:: images/linux/java-11.png
+         :alt: Java 11
+
+  .. tab :: Mac OSX
+
+    **Getting Visual Studio Code**
+
+      VSCode is the IDE (Integrated Development Environment) that is used for
+      2019 and beyond. It needs to be installed on any development computer.
+      It can be downloaded here: https://code.visualstudio.com.
+
+      The downloaded file will be "VSCode-darwin-stable.zip" (1)
+
+      Once downloaded, double-click on the zip file to expand it and copy the
+      new file: "Visual Studio Code" to the Applications folder (2).
+
+      .. figure:: images/mac/VisualStudioCode.png
+         :alt:
+
+    **Download the WPILib release and move the directory**
+
+      Download the software release by navigating to this page:
+      https://github.com/wpilibsuite/allwpilib/releases and downloading the
+      Mac release.
+
+      .. figure:: images/mac/MacReleasePage.png
+         :alt:
+
+      Unzip and untar the file by looking at the file in the explorer and
+      double-clicking on it, once or twice to unzip (remove the .gz extension)
+      and again to untar it (remove the .tar extension). When finished it
+      should like like the folder shown below.
+
+      .. figure:: images/mac/UntarredRelease.png
+         :alt:
+
+      Using Finder (or command line) copy the contents of the folder to a new
+      folder in your home directory, ~/frc2019 as shown below.
+
+      .. figure:: images/mac/MovedFiles.png
+         :alt:
+
+    **Run the ``ToolsUpdater.py`` script**
```

Comment:
This does not render the ToolsUpdater.py properly.

Reply:
Good catch.